### ISSUE TITLE
Inconsistent bin indices in `lookup`

### ISSUE THREAD

**Issue by jl-wynen** [scipp/scipp#2534]
The last line in
```.py
import scipp as sc
import scippneutron as scn

def _with_pulse_time_edges(da: sc.DataArray, dim: str) -> sc.DataArray:
    pulse_time = da.coords[dim]
    one = sc.scalar(1, dtype='int64', unit=pulse_time.unit)
    lo = pulse_time[0] - one
    hi = pulse_time[-1] + one
    mid = sc.midpoints(pulse_time)
    da.coords[dim] = sc.concat([lo, mid, hi], dim)
    return da

vana = scn.load(powgen.data.vanadium_file(),
                advanced_geometry=False,
                load_pulse_times=True,
                mantid_args={'LoadMonitors': False})
ch = vana.attrs['proton_charge'].value.rename(time='pulse_time'))

x = filtering._with_pulse_time_edges(ch >= ch.data.mean(), ch.dim)

l = sc.lookup(x, x.dim)[vana.bins.coords[x.dim]]

vana.bins.coords['good_pulse'] = l
```
raises
```.py
VariableError: Expected
<scipp.Variable> (spectrum: 24794, tof: 1)  index_pair           [None]  [(0, 0), (0, 0), ..., (65836865, 65836865), (65836865, 65836865)], got
<scipp.Variable> (spectrum: 24794, tof: 1)  index_pair           [None]  [(0, 0), (0, 0), ..., (0, 0), (0, 0)].
```
Where the first `index_pair` variable is for `l` and the second for `vana`.
Indices agree for all non-empty bins.

To fix this, `sc.lookup` should return the same bin indices as its input.
Is this always possible? I think it can just copy the input variable.

Note that `sc.bin` returns `(n, n)` for empty bins (like `sc.lookup`). The `(0, 0)`'s in `vana` seem to be specific to `scn.load`.

**Comment by jl-wynen** [comment]
This cannot work for inputs that are a slice into a larger variable. In that case, the input can have its indices refer to arbitrary sections of its buffer. But the output's bins are contiguous in its buffer.

**Comment by SimonHeybrock** [comment]
This works:
```python
vana2 = vana.copy()
vana2.bins.coords['good_pulse'] = l
```
i.e., `copy()` also sets those indices to zero. We therefore have a simpler reproducing example:

```python
vana.bins.coords['tof2'] = vana.bins.coords['tof'].copy()
```

Interestingly, this works:
```python
import scipp as sc

da = sc.data.binned_x(nevent=10, nbin=20)
da.bins.coords['y2'] = da.bins.coords['y'].copy()
```

And so does this:
```python
import scipp as sc
import scippneutron as scn

da = scn.load_nexus(scn.data.get_path('PG3_4844_event.nxs'))
da.bins.coords['tof2'] = da.bins.coords['tof'].copy()
```
indicating that the bug is the `scippneutron.load`, i.e., the conversion from Mantid?

**Comment by jl-wynen** [comment]
> Should sc.bins reject indices that do not comply?

I don't think it should. As it stands, it can be used to change indices to match the use case (like in my original problem).

> but at least our own code should follow this?

Makes sense, as long as it is not too difficult.

What exactly is the problem? I only encountered this in `__setitem__`. Are there other other operations that fail because of it? Maybe the problem is in comparing bin indices. It should look for equivalence, not equality. Don't know how much more work / costly this would be.

**Comment by SimonHeybrock** [comment]
> It should look for equivalence, not equality. Don't know how much more work / costly this would be.

Size-zero bins are the only equivalence case, I think? It is no clear though if we should consider that as equivalent? See https://github.com/scipp/scipp/issues/1967#issuecomment-958680504 for an example of how bins might be resized. That would break if some unknown mechanism in between changed the indices. But maybe this argument is irrelevant if you suggest to do this only for insertion?

**Comment by SimonHeybrock** [comment]
>What exactly is the problem? I only encountered this in __setitem__. Are there other other operations that fail because of it?

This works:
```python
vana.bins.coords['tof'] += vana.bins.coords['tof'].copy()
```
This *is* comparing indices, but it is done by `transform`, and that appears to accept equivalent indices. The difference is that here this is indeed sufficient, and it will also work with completely different indices as long as the bin *sizes* match. The problem is probably specific to `__setitem__`, which inserts a new "table column" into the buffer and has thus more stringent requirements.
